### PR TITLE
refactor: documentation tests

### PR DESCRIPTION
**Pull request by joshuaellis**
### What does it do?

* changes `useReactQuery` to `useDocumentation` (clearer naming)
* sets up plugin's shared MSW server
* uses the `useRBAC` hook instead of `CheckPermissions`
* removes unused dependencies from the `package.json
* changes links to actually be anchors (semantics)
* Adds a much more comprehensive test-suite

### Why is it needed?

* Part of tidying up the test suite to remove the usage of `history` in FE tests
* Better confidence in FE testing

### How to test it?

* AUTOMATED 🪄 

### Related issue(s)/PR(s)

* resolves CONTENT-1497

### CODE REVIEW

**Review by markkaylor — COMMENTED** [review]
Nice, looks like a lot of good improvements! Couple quick questions but I'll test it locally a little later.

File: packages/plugins/documentation/admin/src/hooks/useDocumentation.js
Comment (joshuaellis) [reply packages/plugins/documentation/admin/src/hooks/useDocumentation.js:31]: iirc the `onError` handler of the query won't run if there's a failure in the cache and we've not refetched, so if we use this hook somewhere else (e.g. settings) then we won't know it failed... there was an interesting article by one of the maintainers but im struggling to find it :(

File: packages/plugins/documentation/admin/src/hooks/useDocumentation.js
Comment (joshuaellis) [reply packages/plugins/documentation/admin/src/hooks/useDocumentation.js:28]: I've left this "as is" but passing the actual error is more beneficial – which is what sounds like you're suggesting?

Comment (markkaylor) [reply packages/plugins/documentation/admin/src/hooks/useDocumentation.js:28]: Yes I think passing the actual error would be better if you don't mind updating.

File: packages/plugins/documentation/admin/src/hooks/useDocumentation.js
Comment (markkaylor) [reply packages/plugins/documentation/admin/src/hooks/useDocumentation.js:31]: ok interesting, thanks for letting me know.

**Review by markkaylor — APPROVED** [review]
Tests look good to me, thanks for adding them.

**Review by markkaylor — APPROVED** [review]
Tests look good to me, thanks for adding them.

**Comment by joshuaellis** [comment]
I did the amendment but it dismissed your review 🥲

**Review by markkaylor — APPROVED** [review]

**Pull request merged**
